src/ui/ui_i18n/README.md

```diff
@@ -0,0 +1,325 @@
+```
+
+## React
+
+[React-intl](https://github.com/yahoo/react-intl) library is used for internalization
+React part of the application. It provides React components and an API to format
+dates, numbers, and strings, including pluralization and handling translations.
+
+React Intl uses the provider pattern to scope an i18n context to a tree of components.
+`IntlProvider` component is used to setup the i18n context for a tree. After that we
+are able to use `FormattedMessage` component in order to translate messages.
+`IntlProvider` should wrap react app's root component (inside each react render method).
+
+In order to translate messages we need to pass them into the `IntlProvider`
+from I18n engine:
+
+```js
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { IntlProvider } from 'react-intl';
+
+import i18n from 'i18n';
+
+const locale = i18n.getLocale();
+const messages = i18n.getMessages();
+
+ReactDOM.render(
+  <IntlProvider
+    locale={locale}
+    messages={messages}
+  >
+    <RootComponent>
+      ...
+    </RootComponent>
+  </IntlProvider>,
+  document.getElementById('container')
+);
+```
+
+After that we can use `FormattedMessage` components inside `RootComponent`:
+```js
+import React, { Component } from 'react';
+import { FormattedMessage } from 'react-intl';
+
+class RootComponent extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      name: 'Eric',
+      unreadCount: 1000,
+    };
+  }
+
+  render() {
+    const {
+      name,
+      unreadCount,
+    } = this.state;
+
+    return (
+      <p>
+        <FormattedMessage
+          id="welcome"
+          defaultMessage={`Hello {name}, you have {unreadCount, number} {unreadCount, plural,
+            one {message}
+            other {messages}
+          }`}
+          values={{name: <b>{name}</b>, unreadCount}}
+          />
+          ...
+      </p>
+    );
+  }
+}
+```
+
+## Angular
+
+Angular wrapper has 4 entities: translation `provider`, `service`, `directive`
+and `filter`. Both the directive and the filter use the translation `service`
+with i18n engine under the hood.
+
+The translation `provider` is used for `service` configuration and
+has the following methods:
+- `addMessages(messages: Map<string, string>, [locale: string])` - provides a way to register
+translations with the library
+- `setLocale(locale: string)` - tells the library which language to use by given
+language key
+- `getLocale()` - returns the current locale
+- `setDefaultLocale(locale: string)` - tells the library which language to fallback
+when missing translations
+- `getDefaultLocale()` - returns the default locale
+- `defineFormats(formats: object)` - supply a set of options to the underlying formatter
+
+The translation `service` provides the only one method:
+- `translate(id: string, [{values: object, defaultMessage: string}])` – translate message by id
+
+The translation `filter` has the following syntax:
```

Comment:
__nit:__ please mention where this `filter` is supposed to be used (attributes etc.).